File: client/src/game/conf/TextConfig.ts
class TextConfig {
    private static _dic:Dictionary<TextConfig> = new Dictionary<TextConfig>();
    private static _arr:Array<TextConfig> = new Array<TextConfig>();

	 public Id :number ;
     public Text :string ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<TextConfig>{
        return  TextConfig._dic;
    }
    public static get Arr():Array<TextConfig>{
        return TextConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:TextConfig = <TextConfig>json[i];
				TextConfig._arr.push(config);
				TextConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("TextConfig loader fail!");
        }
    }
}

File: server/src/game/conf/TextConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type TextConfigElem struct {
	Id int
	Text string
	
}

type TextConfig0 struct {
	Dic map[int]*TextConfigElem
	Arr []*TextConfigElem
}

var (
	TextConfig TextConfig0
)

func loadTextConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	TextConfig.Dic = make(map[int]*TextConfigElem)
	TextConfig.Arr = make([]*TextConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &TextConfig.Arr)
	for _,conf := range TextConfig.Arr{
		TextConfig.Dic[conf.Id] = conf
	}
	
}
